json-rpc/src/runtime.rs

```diff
@@ -360,6 +371,7 @@ async fn rpc_request_handler(
     ledger_info: LedgerInfoWithSignatures,
     request_type_label: &str,
     trace_id: String,
+    user_agent: String,
```

Comment:
This could probably be a &str allowing you to avoid a Clone

Reply:
I think it gets passed into a future so I couldn't figure out how to manage the lifetime of it with a ref